src/Type/InterfaceType/ContentNode.php

```diff
@@ -0,0 +1,436 @@
+<?php
+namespace WPGraphQL\Type\InterfaceType;
+
+use GraphQL\Deferred;
+use GraphQL\Type\Definition\ResolveInfo;
+use WPGraphQL\AppContext;
+use WPGraphQL\Data\DataSource;
+use WPGraphQL\Model\Post;
+use WPGraphQL\Model\Term;
+use WPGraphQL\Registry\TypeRegistry;
+
+class ContentNode {
+
+	/**
+	 * Adds the ContentNode Type to the WPGraphQL Registry
+	 *
+	 * @param TypeRegistry $type_registry
+	 */
+	public static function register_type( TypeRegistry $type_registry ) {
+
+		/**
+		 * The Content interface represents Post Types and the common shared fields
+		 * across Post Type Objects
+		 */
+		register_graphql_interface_type(
+			'ContentNode',
+			[
+				'description' => __( 'Nodes used to manage content', 'wp-graphql' ),
+				'resolveType' => function( $post ) use ( $type_registry ) {
+
+					/**
+					 * The resolveType callback is used at runtime to determine what Type an object
+					 * implementing the ContentNode Interface should be resolved as.
+					 *
+					 * You can filter this centrally using the "graphql_wp_interface_type_config" filter
+					 * to override if you need something other than a Post object to be resolved via the
+					 * $post->post_type attribute.
+					 */
+					$type = null;
+
+					if ( isset( $post->post_type ) ) {
+						$post_type_object = get_post_type_object( $post->post_type );
+						if ( isset( $post_type_object->graphql_single_name ) ) {
+							$type = $type_registry->get_type( $post_type_object->graphql_single_name );
+						}
+					}
+
+					return ! empty( $type ) ? $type : null;
+
+				},
+				'fields'      => [
```

Comment:
🤔 I'll have to think through this a bit. At the moment we don't really have multiple interfaces implementing the same fields (other than this one and the Node interface both declaring the "id" field), so sharing fields across many interfaces and types hasn't come up as a pattern to look for and abstract, yet. 

But I can see it becoming maybe more relevant the more we implement interfaces. . .the GraphQL Spec is looking at introducing Interfaces that can implement other Interfaces though, so that might solve the issue of sharing fields across many interfaces too.

We could even provide that functionality on our own via the registry, so we wouldn't have to re-define fields. 

Like the `ContentNode` Interface could `implement` the `Node` Interface which would add the ID field to it so that we didn't have to re-define the `ID` field again for `contentNode`

🤔